umh-core/cmd/main.go

```diff
@@ -53,13 +54,61 @@ func main() {
 
 	// Load the config
 	configManager := config.NewFileConfigManager()
-	config, err := configManager.GetConfig(ctx, 0)
+	configData, err := configManager.GetConfig(ctx, 0)
 	if err != nil {
 		sentry.ReportIssuef(sentry.IssueTypeFatal, log, "Failed to load config: %s", err)
 	}
 
+	// Extract environment variables and apply them to the config
+	configModified := false
+
+	// Extract AUTH_TOKEN from environment if present
+	authToken, err := env.GetAsString("AUTH_TOKEN", false, "")
```

Comment:
the logic here could be that without the authToken the communication module is simply not started